### PR TITLE
UHF-2879: Style 404, 403 and 401 pages, add illustrations for error a…

### PR DESCRIPTION
**Pull request by teroelonen**
…nd login pages and maintenance page, add preprocess check if we are on error page and add class to body if so.

How to test:

1. Take any instance and set it up with latest dev branch.
2. `composer require drupal/hdbt:dev-UHF-2879_error_pages`
3. `make drush-cr`
4. `drush locale:check && drush locale:update`
5. Go to a url that doesn't exist. For example '/asfdgfkhlgutjfhgkj' is a good one. The 404 page should look nice now.
6. Log out and try to go to '/admin' for example. That 403 page should look nice as well. It should be 1:1 same as 401 page. If you know how to get into 401 page you can check it too :)
7. Test all error pages on all languages as well.

Attention! the login block on 403 & 401 page doesn't look similar to the layouts since it will be styled to the designed look on story https://helsinkisolutionoffice.atlassian.net/browse/UHF-3922

### CODE REVIEW

File: src/scss/layout/_error-pages.scss [hunk 1]
```diff
@@ -0,0 +1,91 @@
+.error-page {
```
Comment (Arkkimaagi) [inline src/scss/layout/_error-pages.scss:1]: There is no need for this nesting. You're already taking care of it with the namespacing of BEM. Please do not add extra specificity where it's not needed.

File: src/scss/layout/_error-pages.scss [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      width: 50%;
+    }
+  }
+
+  .error-page__title {
+    margin-bottom: 0;
+  }
+
+  .error-page__description {
+    // TODO: change to use correct font-definition once the PR is merged. Change to body.
```
Comment (Arkkimaagi) [inline src/scss/layout/_error-pages.scss:27]: Well done on this thinking ahead! Good job!

**Comment by Arkkimaagi** [comment]
I tried finding 401 page, I even asked our drupal gurus. Apparently Drupal only uses 401 with API requests.

> This does make sense once you consider that anonymous users can also have permissions. So from Drupal's side the problem isn't that you're not logged in, its that your anonymous user account doesn't have the correct permissions.

So maybe we should just remove the 401 template from confusing us in the future?

File: src/scss/layout/_error-pages.scss (hunk 1 above)
Comment (teroelonen) [reply src/scss/layout/_error-pages.scss:1]: Ah good catch! This was actually a mistake :D only the .no-sidebar .main-content was supposed to be nested. Fixed.

File: src/scss/layout/_error-pages.scss (hunk 2 above)
Comment (teroelonen) [reply src/scss/layout/_error-pages.scss]: Thanks :)

**Comment by teroelonen** [comment]
The 401 page is something that is really rare I heard too :) However I don't think it should be removed just because. The template is named so that everyone knows what this page is for. I would keep it just to be safe and its 1:1 403 anyway.

**Review by Arkkimaagi — CHANGES_REQUESTED** [review]

File: templates/layout/page--401.html.twig [hunk 3]
```diff
@@ -1,18 +1,28 @@
...
+        <div class="error-page__text-container">
+          <div class="page-title">
+            <h1 class="error-page__title">403 - {{ 'You do not have permission to access this page'|t }}</h1>
+          </div>
+          <p class="error-page__description">{{ 'This page is available to registered users only. If you have the necessary permissions, you can sign in to see the content.'|t }}</p>
+          <a href="{{ error_page_home_link }}" class="error-page__link">{{ 'Go back to the hel.fi homepage'|t({}, {'context': 'Return to homepage link for error pages'}) }} {% include '@hdbt/misc/icon.twig' with {icon: 'arrow-right'} %}</a><br/>
+          <a href="{{ error_page_feedback_link }}" class="error-page__link">{{ 'Give feedback'|t({}, {'context': 'Feedback link for error pages'}) }} {% include '@hdbt/misc/icon.twig' with {icon: 'arrow-right'} %}</a>
+          {% if user_not_logged_in %}
+            <h2 class="error-page__login-title">{{ 'Log in'|t({}, {'context': 'Log in block title on error pages'}) }}</h2>
+          {% endif %}
```
[the remark this answers is not in the document]
Comment (teroelonen) [reply templates/layout/page--401.html.twig:14]: No. It is printed only if the user has access to it so it has the functionality built in already.

File: hdbt.theme [hunk 4]
```diff
@@ -173,6 +185,29 @@ function hdbt_get_settings($setting, $group = 'site_settings') {
  * Implements hook_preprocess_HOOK().
  */
 function hdbt_preprocess_page(&$variables) {
+  // Check what language is set and provide links to homepage and feedback
+  // accordingly.
+  switch ($variables['current_langcode']) {
+    case 'fi':
+      $variables['error_page_home_link'] = 'https://www.hel.fi/helsinki/fi';
+      $variables['error_page_feedback_link'] = 'https://www.hel.fi/helsinki/fi/kaupunki-ja-hallinto/osallistu-ja-vaikuta/palaute';
```
[the remark this answers is not in the document]
Comment (teroelonen) [reply hdbt.theme:193]: They actually might link to different feedback links (there is multiple on the hel.fi site) so there is no "one truth".

File: templates/layout/page--401.html.twig [hunk 5]
```diff
@@ -1,18 +1,28 @@
 {% embed 'page.html.twig' %}
-  {% block page_before_content %}
-    {% embed '@hdbt/misc/container.twig' with {container_element: 'page-title'} %}
+  {% block page_content %}
```
[the remark this answers is not in the document]
Comment (teroelonen) [reply templates/layout/page--401.html.twig:2]: 🙇 ❤️

File: src/scss/layout/_error-pages.scss [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+  .error-page__login-form {
+    margin-top: $spacing-double;
+  }
+
+  .error-page__illustration-container {
+    @include breakpoint($breakpoint-m) {
+      align-items: flex-start;
+      display: flex;
+      justify-content: center;
+      padding: $spacing-sextuple $spacing-triple 0 $spacing-triple;
```
[the remark this answers is not in the document]
Comment (teroelonen) [reply src/scss/layout/_error-pages.scss]: Fixed.

File: templates/layout/page--401.html.twig [hunk 7]
```diff
@@ -1,18 +1,28 @@
...
+    {% embed '@hdbt/misc/container.twig' with {container_element: 'error-page-content'} %}
       {% block container_content %}
-        <div class="page-title">
-          <h1>401 - {{ 'You are unauthorized to view this page.'|t }}</h1>
+        <div class="error-page__text-container">
+          <div class="page-title">
+            <h1 class="error-page__title">403 - {{ 'You do not have permission to access this page'|t }}</h1>
```
[the remark this answers is not in the document]
Comment (teroelonen) [reply templates/layout/page--401.html.twig]: Good catch, fixed.

File: translations/fi.po [hunk 8]
```diff
@@ -85,3 +85,27 @@ msgstr "Lisätietolinkit"
...
+msgstr "Sivu näkyy vain kirjautuneille käyttäjille. Jos sinulla on riittävät käyttöoikeudet, näet sisällön kirjautumalla sisään."
+
+msgid "Sorry, the page you were looking for was not found"
+msgstr "Etsimääsi sivua ei valitettavasti löytynyt"
+
+msgid "Check that you have typed in the page address correctly. We may have deleted or moved the content."
+msgstr "Tarkista, että sivun osoite on kirjoitettu oikein. Olemme myös saattaneet poistaa tai siirtää sivun."
+
+msgctxt "Return to homepage link for error pages"
+msgid "Go back to the hel.fi homepage"
```
[the remark this answers is not in the document]
Comment (teroelonen) [reply translations/fi.po:102]: Changed all to lowercase - copy paste from ticket so maybe it came from there.

File: src/images/illustration_error_page_403_401.svg [hunk 9]
```diff
@@ -0,0 +1,535 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (teroelonen) [reply src/images/illustration_error_page_403_401.svg:1]: Maybe some day - Markus tried quickly with webpack to set it up but it didn't work so maybe create a ticket out this?

**Comment by Arkkimaagi** [comment]
Works fine, and is updated to latest coding styles, great job! Approved 🏁

**Review by Arkkimaagi — APPROVED** [review]

**Pull request merged**